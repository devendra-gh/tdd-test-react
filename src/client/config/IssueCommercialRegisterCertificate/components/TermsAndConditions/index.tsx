import React from 'react';
import { IVariables } from '@tamm/app-composer';
import Checkbox from '@tamm/ui-lib-v2-checkbox';

function TermsAndConditions(props: IVariables) {
  const { items, i18n, checked, handleClick, showErrors } = props;

  return items.map((item: IVariables, index: number) => (
    <Checkbox
      {...item}
      label={i18n(item.label)}
      checked={checked[index]}
      validateStatus={showErrors && !checked[index] ? 'error' : 'success'}
      onClick={() => handleClick(index)}
    />
  ));
}

export default TermsAndConditions;
