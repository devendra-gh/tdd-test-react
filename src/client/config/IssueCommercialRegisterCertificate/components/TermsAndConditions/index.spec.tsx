import React from 'react';
import { shallow } from 'enzyme';
import TermsAndConditions from './index';

jest.mock('@tamm/app-composer', () => ({
  withTemplateHooks: (component: any) => component,
}));

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('Select Licence template', () => {
  let props: any;
  beforeEach(() => {
    props = {
      i18n: jest.fn(),
      handleClick: jest.fn(),
      checked: [true],
      showErrors: false,
      items: [
        {
          label: 'test label',
        },
      ],
    };
  });
  it('should successfully render', () => {
    const component = shallow(<TermsAndConditions {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should should show errors when term is not accepted and vaildation has run', () => {
    props.showErrors = true;
    props.checked = [false];
    const component = shallow(<TermsAndConditions {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should call the handle click function on click', () => {
    const component = shallow(<TermsAndConditions {...props} />);
    component.props().onClick();
    expect(props.handleClick).toHaveBeenCalled();
  });
});
