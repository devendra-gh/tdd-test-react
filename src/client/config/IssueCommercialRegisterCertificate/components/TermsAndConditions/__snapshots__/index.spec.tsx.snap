// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Licence template should should show errors when term is not accepted and vaildation has run 1`] = `
<Checkbox
  checked={false}
  onClick={[Function]}
  validateStatus="error"
/>
`;

exports[`Select Licence template should successfully render 1`] = `
<Checkbox
  checked={true}
  onClick={[Function]}
  validateStatus="success"
/>
`;
